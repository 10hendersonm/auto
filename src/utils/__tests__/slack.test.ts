import postToSlack from '../slack';

const fetchSpy = jest.fn();
// @ts-ignore
jest.mock('node-fetch', () => (...args) => {
  fetchSpy(...args);
});

beforeEach(() => {
  fetchSpy.mockClear();
});

describe('postToSlack', () => {
  test('should throw without a token', async () => {
    expect(
      postToSlack('# My Notes', {
        slackUrl: 'https://custom-slack-url',
        baseUrl: 'https://github.custom.com',
        owner: 'Adam Dierkens',
        repo: 'test',
        tag: '1.0.0'
      })
    ).rejects.toEqual(new Error('Slack needs a token to send a message'));
  });

  test('should call slack api', async () => {
    process.env.SLACK_TOKEN = 'MY_TOKEN';

    await postToSlack('# My Notes\n- PR [some link](google.com)', {
      slackUrl: 'https://custom-slack-url',
      baseUrl: 'https://github.custom.com',
      owner: 'Adam Dierkens',
      repo: 'test',
      tag: '1.0.0'
    });

    expect(fetchSpy).toHaveBeenCalled();
    expect(fetchSpy.mock.calls[0][0]).toBe(
      'https://custom-slack-url?token=MY_TOKEN'
    );
    expect(fetchSpy.mock.calls[0][1].body).toMatchSnapshot();
  });
});
