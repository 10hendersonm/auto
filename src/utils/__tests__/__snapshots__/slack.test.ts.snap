// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postToSlack should call slack api 1`] = `"{\\"text\\":\\"@channel: New release *1.0.0*\\\\nhttps://github.custom.com/Adam Dierkens/test/releases/tag/1.0.0\\\\n\\\\n\`\`\`# My Notes\\\\n- PR some link\`\`\`\\",\\"link_names\\":1}"`;
